x/distribution/abci_app.go

```diff
@@ -32,8 +19,15 @@ func getPreviousPercentPrecommitVotes(req abci.RequestBeginBlock) sdk.Dec {
 		}
 	}
 
-	if totalPower == 0 {
-		return sdk.ZeroDec()
+	// TODO this is Tendermint-dependent
+	// ref https://github.com/cosmos/cosmos-sdk/issues/3095
+	if ctx.BlockHeight() > 1 {
+		previousProposer := k.GetPreviousProposerConsAddr(ctx)
+		k.AllocateTokens(ctx, sumPrecommitPower, totalPower, previousProposer, req.LastCommitInfo.GetVotes())
 	}
-	return sdk.NewDec(sumPrecommitPower).Quo(sdk.NewDec(totalPower))
+
+	// record the proposer for when we payout on the next block
+	consAddr := sdk.ConsAddress(req.Header.ProposerAddress)
+	k.SetPreviousProposerConsAddr(ctx, consAddr)
```

Comment:
Its silly that distribution has to keep track of this :(, another thing that should "just be known" within the SDK / context.

Reply:
Yeah, and from what it seems to me, this is only used for `ExportGenesis` and no where else, in which case we definitely shouldn't be storing it in state.  Should just be accessible through `ctx` or something.

Reply:
Completely agree here.

Reply:
@sunnya97 We use it each block to determine which proposer gets the proposer reward, so we definitely do need to store it somewhere.

Reply:
Made an issue - https://github.com/cosmos/cosmos-sdk/issues/3273